streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -435,25 +451,28 @@ public void process(final ProcessorSupplier<K, V> processorSupplier, String... s
      * an operation that changes the key, i.e, selectKey, map(..), flatMap(..).
      * @param keySerde      Serdes for serializing the keys
      * @param valSerde      Serdes for serilaizing the values
+     * @param storeName     state store name for source to be repartitioned
      * @return a new {@link KStreamImpl}
      */
     private KStreamImpl<K, V> repartitionForJoin(Serde<K> keySerde,
-                                                 Serde<V> valSerde) {
+                                                 Serde<V> valSerde,
+                                                 final String storeName) {
```

Comment:
Nit: change the parameter name for `repartitionForJoin` and `createReparitionedSource` as `topicNamePrefix` since repartitioning is not necessarily followed by materialization.